src/components/Divider.tsx

```diff
@@ -1,61 +1,141 @@
+    const isTableView =
+      (parentId && attributeEquals(state, parentId, '=view', 'Table')) ||
+      (grandParentId && attributeEquals(state, grandParentId, '=view', 'Table')) ||
+      false
+
+    let thoughtsAtSameDepth: { id: ThoughtId; value: string }[] = []
+
+    if (isTableView && isOnlyChild && grandParentId) {
+      const parentSiblings = getAllChildrenAsThoughts(state, grandParentId)
+      thoughtsAtSameDepth = parentSiblings.flatMap(parent => {
+        const childrenOfParent = parent.id ? getAllChildrenAsThoughts(state, parent.id) : []
+        return childrenOfParent.filter(child => !isDivider(child.value))
+      })
+    }
+
+    return {
+      dividerId,
+      parentId,
+      isOnlyChild,
+      isTableView,
+      children,
+      thoughtsAtSameDepth,
+    }
+  })
+}
+
+/** Helper function to get widths of thoughts by querying the DOM. */
+const getThoughtWidths = (thoughts: { id: ThoughtId }[]) => {
+  return thoughts.map(thought => {
+    const innerThoughtElement = document.querySelector(
+      `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+    )
+    return innerThoughtElement ? innerThoughtElement.getBoundingClientRect().width : 0
+  })
+}
 
 /** A custom horizontal rule. */
 const Divider = ({ path, cssRaw }: { path: Path; cssRaw?: SystemStyleObject }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const [dividerWidth, setDividerWidth] = useState<number>(DIVIDER_MIN_WIDTH)
+
+  // Use the custom hook to get necessary data
+  const { dividerId, isOnlyChild, isTableView, children, thoughtsAtSameDepth } = useDividerData(path)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
-  const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.querySelector('[data-divider=true]')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
+  useLayoutEffect(() => {
+    /** Calculates and updates the Divider's width based on sibling thought widths. */
+    const updateDividerWidth = () => {
+      if (!dividerRef.current) return
+
+      // If the divider is the only child and we're not in Table View
```

Comment:
Not a blocking issue, but this comment is unnecessary. Since the variables are clearly named, the comment is redundant and does not add additional information.

`if (isOnlyChild && !isTableView)` reads somewhat literally as `if is-only-child and is-not-table-view`, which is almost identical to the comment